app/jinja_filters.py

```diff
@@ -75,12 +75,18 @@ def format_unit(
     value: int | float | Decimal,
     length: UnitLengthType = "short",
 ) -> str:
+    # mass-metric-ton no longer supported for en_GB and related locales, but still present in business schema and allowed in validator,
+    # until removed from schema we substitute mass-ton for mass-metric-ton and vice versa, before/after unit formatting
+    measurement_unit = "mass-ton" if unit == "mass-metric-ton" else unit
+
     formatted_unit: str = units.format_unit(
         value=value,
-        measurement_unit=unit,
+        measurement_unit=measurement_unit,
         length=length,
         locale=flask_babel.get_locale(),
     )
+    if unit == "mass-metric-ton":
+        return formatted_unit.replace("tn", "mt").replace("ton", "metric ton")
```

Comment:
I installed babel 2.11.0 and ran 
```
print(units.format_unit(1, "mass-metric-ton", locale=Locale('en', 'GB')))
print(units.format_unit(1, "mass-metric-ton", length="short", locale=Locale('en', 'GB')))
```
which outputted 
`1 tonne`
`1 t`
these replacements aren't quite what it used to be

Reply:
(see other comment) https://github.com/ONSdigital/eq-questionnaire-runner/pull/1146#discussion_r1242015494

Reply:
But we use 2.12.1 already.

Reply:
Yep, so we've upgraded to 2.12.1 which has broken the schema, because we can't call format unit with `mass-metric-ton` anymore, so I installed the most recent version of babel that did still support this and checked what the output would be, which is `tonne` and `t`
So if it was changed to `metric-ton` now, this is different to what users would have seen when we were using a prior version of babel where this schema worked

Reply:
Ok, I get it now, so it also adds to the confusion as in schema we are using metric tons but want tonne/t output in runner.

Reply:
yeah looking at `qfs_0002` it uses `tonnes` in the description but `mass-metric-ton` so that would mirror this too
```
"label": "Total quantity of heavy fuel oil purchased in tonnes",
"description": "Please provide your figures rounded to the nearest whole number. Enter &#39;0&#39;, rather than leave blank.",
"q_code": "122",
"unit": "mass-metric-ton",
"unit_length": "short",
"decimal_places": 0
```

Reply:
I will double check what you did with 2.11 but these are terrific simplifications thanks tom your findings!